php/libraries/Database.class.inc

```diff
@@ -882,14 +892,17 @@ class Database
      * Runs a query as a prepared statement and returns the value of the first
      * column of the first row
      *
-     * @param string $query  The SQL statement to run
-     * @param array  $params Values to use for binding in the prepared statement
+     * @param string $query           The SQL statement to run
+     * @param array  $params          Values to use for binding in the prepared
+     *                                statement
+     * @param bool   $efficiencyLimit Determines whether to use LIMIT clause in
+     *                                pselectRow
      *
      * @return mixed The value returned by the query.
      */
-    function pselectOne($query, $params)
+    function pselectOne($query, $params, $efficiencyLimit = true)
     {
-        $result = $this->pselectRow($query, $params);
+        $result = $this->pselectRow($query, $params, $efficiencyLimit);
```

Comment:
should not be needed as pselectRow implement a hard coded limit of 1and you always want the most efficient query in you database.